py_zipkin/zipkin.py

```diff
@@ -206,16 +206,17 @@ def start(self):
         # If zipkin_attrs are passed in or this span is doing its own sampling,
         # it will need to actually log spans at __exit__.
         self.perform_logging = self.zipkin_attrs or self.sample_rate is not None
+        report_root_timestamp = False
 
         if self.sample_rate is not None:
-            # This clause allows for sampling this service independently
-            # of the passed-in zipkin_attrs.
             if self.zipkin_attrs and not self.zipkin_attrs.is_sampled:
```

Comment:
don't let this confuse you, but in the java impl, we separate out extracting data from the headers, from joining a span or creating a new trace.

```java
contextOrFlags = extractor.extract(request); // the B3 header parser doesn't know if the context will be shared
span = contextOrFlags.context() != null
  ? tracer.joinSpan(contextOrFlags.context()) // this sets shared = true
  : tracer.newTrace(contextOrFlags.samplingFlags()); // this leaves shared = false
```